frontend/src/metabase-lib/utils/create-lookup-by-property.ts

```diff
@@ -1,8 +1,8 @@
-export function createLookupByProperty(items: any[], property: string) {
-  const lookup: Record<string, any> = {};
+export function createLookupByProperty<T>(items: T[], property: keyof T) {
```

Comment:
Fix types so that it infers a better return value type. Previously id would return `Record<string, any>`